examples/runners/ash/src/device.rs

```diff
@@ -0,0 +1,185 @@
+                    .enumerate()
+                    .find(|(_, prop)| prop.queue_flags.contains(vk::QueueFlags::GRAPHICS))
+                    .ok_or(anyhow!(
+                        "No graphics + compute queues on physical device available"
+                    ))?
+                    .0 as u32
+            };
+
+            let device = instance
+                .create_device(
+                    physical_device,
+                    &vk::DeviceCreateInfo::default()
+                        .push_next(
+                            &mut vk::PhysicalDeviceVulkan12Features::default()
+                                .vulkan_memory_model(true),
+                        )
+                        .push_next(
+                            &mut vk::PhysicalDeviceVulkan13Features::default()
+                                .synchronization2(true)
+                                .dynamic_rendering(true),
+                        )
+                        .queue_create_infos(&[vk::DeviceQueueCreateInfo::default()
+                            .queue_family_index(main_queue_family)
+                            .queue_priorities(&[1.0])])
+                        .enabled_extension_names(&[
+                            khr::swapchain::NAME.as_ptr(),
+                            khr::shader_non_semantic_info::NAME.as_ptr(),
+                        ]),
+                    None,
+                )
+                .context("create_device")?;
+            let main_queue = device.get_device_queue(main_queue_family, 0);
+
+            Ok(Arc::new(Self {
+                debug_ext_device: ext::debug_utils::Device::new(&instance, &device),
+                surface_ext: khr::surface::Instance::new(&entry, &instance),
+                swapchain_ext: khr::swapchain::Device::new(&instance, &device),
+                entry,
+                instance,
+                physical_device,
+                device,
+                main_queue_family,
+                main_queue,
+                debug_ext_instance: debug_instance,
+                debug_callback,
+            }))
+        }
+    }
+}
+
+impl Drop for MyDevice {
+    fn drop(&mut self) {
+        unsafe {
+            self.debug_ext_instance
+                .destroy_debug_utils_messenger(self.debug_callback, None);
+            self.device.destroy_device(None);
+            self.instance.destroy_instance(None);
+        }
+    }
+}
+
+unsafe extern "system" fn vulkan_debug_callback(
+    message_severity: vk::DebugUtilsMessageSeverityFlagsEXT,
+    _message_type: vk::DebugUtilsMessageTypeFlagsEXT,
+    p_callback_data: *const vk::DebugUtilsMessengerCallbackDataEXT<'_>,
+    _user_data: *mut std::os::raw::c_void,
+) -> vk::Bool32 {
+    let callback_data = unsafe { *p_callback_data };
+    let message_id_name = if callback_data.p_message_id_name.is_null() {
+        Cow::from("")
+    } else {
+        unsafe { CStr::from_ptr(callback_data.p_message_id_name).to_string_lossy() }
+    };
```

Comment:
https://docs.rs/ash/0.38.0/ash/vk/struct.DebugUtilsMessengerCallbackDataEXT.html#method.message_id_name_as_c_str would this help?